pkg/daemon/update.go

```diff
@@ -245,16 +247,45 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 		}
 	}()
 
+	if err := dn.updateFIPS(oldConfig, newConfig); err != nil {
+		return err
+	}
+	defer func() {
+		if retErr != nil {
+			if err := dn.updateFIPS(newConfig, oldConfig); err != nil {
+				retErr = errors.Wrapf(retErr, "error rolling back FIPS %v", err)
+				return
+			}
+		}
+	}()
+
 	return dn.updateOSAndReboot(newConfig)
 }
 
+func (dn *Daemon) updateFIPS(current, desired *mcfgv1.MachineConfig) error {
+	if current.Spec.Fips == desired.Spec.Fips {
```

Comment:
Does the current get written to Status anywhere?

Reply:
nope